modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-channel-test/src/testIntegration/java/com/liferay/headless/commerce/admin/channel/resource/v1_0/test/PaymentMethodGroupRelOrderTypeResourceTest.java

```diff
@@ -15,15 +15,300 @@
+				expirationDateConfig.getMonth(), expirationDateConfig.getDay(),
+				expirationDateConfig.getYear(), expirationDateConfig.getHour(),
+				expirationDateConfig.getMinute(), true, _serviceContext);
+
+		_commerceOrderTypes.add(commerceOrderType);
+
+		return commerceOrderType;
+	}
+
+	private CommercePaymentMethodGroupRelQualifier
+			_addCommercePaymentMethodGroupRelQualifier(
+				CommerceOrderType commerceOrderType,
+				long commercePaymentMethodGroupRelId)
+		throws Exception {
+
+		CommercePaymentMethodGroupRelQualifier
+			commercePaymentMethodGroupRelQualifier =
+				_commercePaymentMethodGroupRelQualifierLocalService.
+					addCommercePaymentMethodGroupRelQualifier(
+						_user.getUserId(), CommerceOrderType.class.getName(),
+						commerceOrderType.getCommerceOrderTypeId(),
+						commercePaymentMethodGroupRelId);
+
+		_commercePaymentMethodGroupRelQualifiers.add(
+			commercePaymentMethodGroupRelQualifier);
+
+		return commercePaymentMethodGroupRelQualifier;
+	}
+
+	private PaymentMethodGroupRelOrderType _addPaymentMethodGroupRelOrderType(
+			Long commercePaymentMethodGroupRelId)
+		throws Exception {
+
+		CommerceOrderType commerceOrderType = _addCommerceOrderType();
+
+		CommercePaymentMethodGroupRelQualifier
+			commercePaymentMethodGroupRelQualifier =
+				_addCommercePaymentMethodGroupRelQualifier(
+					commerceOrderType, commercePaymentMethodGroupRelId);
+
+		_commercePaymentMethodGroupRelQualifiers.add(
+			commercePaymentMethodGroupRelQualifier);
+
+		return new PaymentMethodGroupRelOrderType() {
+			{
+				orderTypeExternalReferenceCode =
+					commerceOrderType.getExternalReferenceCode();
+				orderTypeId = commerceOrderType.getCommerceOrderTypeId();
+				paymentMethodGroupRelId =
+					commercePaymentMethodGroupRelQualifier.
+						getCommercePaymentMethodGroupRelId();
+				paymentMethodGroupRelOrderTypeId =
+					commercePaymentMethodGroupRelQualifier.
+						getCommercePaymentMethodGroupRelQualifierId();
+			}
+		};
+	}
+
+	private long _getId() throws Exception {
+		CommercePaymentMethodGroupRel commercePaymentMethodGroupRel =
+			CommercePaymentMethodGroupRelLocalServiceUtil.
+				addCommercePaymentMethodGroupRel(
+					_user.getUserId(), _commerceChannel.getGroupId(),
+					RandomTestUtil.randomLocaleStringMap(),
+					RandomTestUtil.randomLocaleStringMap(), null,
+					RandomTestUtil.randomString(), 99, true);
+
+		_commercePaymentMethodGroupRels.add(commercePaymentMethodGroupRel);
+
+		return commercePaymentMethodGroupRel.
+			getCommercePaymentMethodGroupRelId();
+	}
+
+	@DeleteAfterTestRun
+	private CommerceChannel _commerceChannel;
+
+	@DeleteAfterTestRun
+	private CommerceCurrency _commerceCurrency;
+
+	@DeleteAfterTestRun
+	private List<CommerceOrderType> _commerceOrderTypes = new ArrayList<>();
+
+	@Inject
+	private CommercePaymentMethodGroupRelQualifierLocalService
+		_commercePaymentMethodGroupRelQualifierLocalService;
+
+	@DeleteAfterTestRun
```

Comment:
Are you sure this also works with collections?

Reply:
yes I've seen it used in other test classes too